hack/generate-manifest.sh

```diff
@@ -192,6 +192,14 @@ if [[ $TUN_TYPE != "geneve" ]]; then
     sed -i.bak -E "s/^[[:space:]]*#[[:space:]]*tunnelType[[:space:]]*:[[:space:]]*[a-z]+[[:space:]]*$/tunnelType: $TUN_TYPE/" antrea-agent.conf
 fi
 
+if [[ $CLOUD != "" ]]; then
+    # Delete the serviceCIDR parameter for the cloud (AKS, EKS, GKE) deployment yamls, because
+    # AntreaProxy is always enabled for the cloud managed K8s clusters, and the serviceCIDR
+    # parameter is not needed in this case.
+    sed  -i.bak ':a;N;s/\n/&/4;Ta;/#serviceCIDR:/s/.*\n//;P;D' antrea-agent.conf
+    sed  -i.bak '/#serviceCIDR:/d' antrea-agent.conf
```

Comment:
I tested this one but cannot figure out what the one above is for? I assume to delete the comment maybe but it doesn't seem to do anything on macOS.

Reply:
Yes, it is to remove the comments (by removing the 4 lines before the match).

If it does not work on MAC, it must be due to sed behaviors. But do you have any better solution?

Reply:
We need a sed command that works on both OSes, but I may have tested this one wrong. I develop on macOS directly and I think others do a well.

Reply:
Let me try on my MAC too (but mine is of an older version).

Reply:
I changed the sed commands and tested on both Ubuntu and macOS.